tests/models/test_cpu.py

```diff
@@ -202,22 +250,38 @@ def test_running_test_after_fitting(tmpdir):
         limit_train_batches=0.4,
         limit_val_batches=0.2,
         limit_test_batches=0.2,
-        checkpoint_callback=checkpoint,
+        callbacks=[checkpoint],
         logger=logger,
     )
     result = trainer.fit(model)
 
-    assert result == 1, 'training failed to complete'
+    assert result == 1, "training failed to complete"
 
     trainer.test()
 
     # test we have good test accuracy
-    tutils.assert_ok_model_acc(trainer, thr=0.5)
+    tutils.assert_ok_model_acc(trainer, key='test_loss', thr=0.5)
 
 
 def test_running_test_no_val(tmpdir):
     """Verify `test()` works on a model with no `val_loader`."""
-    model = EvalModelTemplate()
+    class ModelTrainTest(BoringModel):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+        def val_loader(self):
+            pass
+
+        def test_step(self, batch, batch_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            return {"y": loss}
+
+        def test_epoch_end(self, outputs) -> None:
+            test_loss = torch.stack([x["y"] for x in outputs]).mean()
+            self.log('test_loss', test_loss)
```

Comment:
```suggestion
        def test_step(self, batch, batch_idx):
            output = self.layer(batch)
            loss = self.loss(batch, output)
            self.log('test_loss', loss)
```
same here.

Reply:
I discussed with @Borda about BoringModel, and we decided to keep the model as minimal as possible. Specifically, if a test need an additional functionality (e.g., set the learning rate, set the optimizer, aggregate the return values and epoch_end), it should inherit the basic BoringModel in a subclass that implements these features. For examples, logging is tested in `test_hparams`, optimizers and lr are tested in `test_cpu`, and so on. This also grantee more compatibility with exisiting tests. If I misinterpreted @Borda 's suggestion, I can fix these!

Reply:
If BordingModel should be kept minimal, I'd suggest removing `self.optimizer`, `self.batch_size` and `self.learning_rate` too. If required one should override these methods wherever it's required.

Regarding the methods, `self.log()` with `on_epoch=True`(default) in `test_step` will aggregate the results for you. No need to override `epoch_end`. Same for validation_step.

Reply:
Regarding the parameters, I proposed a solution that doesn't require input parameters, while still defining parameters instead of using them as "magic constants". I believe this solution should be clean enough, maybe @Borda can help us on this matter.

On the other hand, `self.log()` introduce a new behavior to `BoringModel` that wasn't conceived, and therefore if I add the logging of the aggregation results other tests (e.g., in `tests/checkpointing`) fail. I'd prefer to keep this PR simple, and add a TODO for introducing epoch_end methods in BoringModel in a different PR

Reply:
ok, I see, boring model already has `_epoch_end` methods overridden. It's ok then. Or you can log it in val/test_step, still return the dict just for the sake of the epoch_end methods already defined in the inherited BoringModel.

TODO is not required BTW.

Reply:
I would prefer have no logging by default in the BoringModel.

Reply:
@tchaton I agree, in fact I removed logging. You can check the most recent commit for this, in the mean time I resolved this conversation

Cheers! :)